### DISCUSSION TITLE
Bidirect client example

### DISCUSSION THREAD

**Discussion by blip-machine418** [connectrpc/connect-es#1284] [Q&A] [answered]
Did someone implemented bidirect client using createPromiseClient?

**Comment by smaye81** [comment, marked as answer]
Hi there. For an example of bidirectional streaming with the promise client, you can check our Connect conformance tests here: https://github.com/connectrpc/connect-es/blob/main/packages/connect-conformance/src/promise-client.ts#L185

The conformance tests are a suite of tests that verify conformance with the Connect protocol. The linked code above shows an example of bidirectional streaming, but in that file you can find examples of all the various kinds of RPCs. Just keep in mind that bidirectional streaming is not currently possible from a browser. If you are implementing this for a browser environment, you will be limited to unary or server-streaming requests.
